pkg/ingester/otlp/convert.go

```diff
@@ -198,16 +249,30 @@ func (p *profileBuilder) convertFunctionBack(of *otelProfile.Function) uint64 {
 	return gf.Id
 }
 
-func (p *profileBuilder) convertSampleBack(os *otelProfile.Sample) *googleProfile.Sample {
+func (p *profileBuilder) convertSampleBack(os *otelProfile.Sample) (*googleProfile.Sample, error) {
 	gs := &googleProfile.Sample{
 		Value: os.Value,
 	}
-
 	if len(gs.Value) == 0 {
-		gs.Value = []int64{int64(len(os.TimestampsUnixNano))}
-	} else if len(gs.Value) == 1 && p.cpuConversion {
-		gs.Value[0] *= p.src.Period
+		return nil, fmt.Errorf("sample value is required")
 	}
+
+	for i, typ := range p.sampleProcessingTypes {
+		switch typ {
+		case sampleConversionTypeSamplesToNanos:
+			gs.Value[i] *= p.src.Period
+		case sampleConversionTypeSumEvents:
```

Comment:
Do you think adding a comment for the future us saying something like the following would help (feel free to omit it):
```
// For off-CPU profiles, aggregate all sample values into a single sum
// since pprof cannot represent variable-length sample values
```